src/h5web/vis-packs/core/shared/Html.tsx

```diff
@@ -4,38 +4,82 @@ import {
   forwardRef,
   HTMLAttributes,
   useLayoutEffect,
+  useRef,
+  PropsWithChildren,
+  ForwardedRef,
 } from 'react';
 import ReactDOM from 'react-dom';
-import { useThree } from '@react-three/fiber';
+import { useFrame, useThree } from '@react-three/fiber';
+import { Camera, Group, Object3D, Vector3 } from 'three';
 
 // Simplified version of `drei`'s `<Html>` component
-// https://github.com/pmndrs/drei/blob/v2.2.3/src/Html.tsx
-const Html = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
-  (props, ref) => {
-    const { className, style, children, ...divProps } = props;
+// https://github.com/pmndrs/drei/blob/v6.0.3/src/web/Html.tsx
 
-    const { width, height } = useThree((state) => state.size);
+const v1 = new Vector3();
+
+function defaultCalculatePosition(
+  el: Object3D,
+  camera: Camera,
+  size: { width: number; height: number }
+) {
+  const objectPos = v1.setFromMatrixPosition(el.matrixWorld);
+  objectPos.project(camera);
+  const widthHalf = size.width / 2;
+  const heightHalf = size.height / 2;
+  return [
+    objectPos.x * widthHalf + widthHalf,
+    -(objectPos.y * heightHalf) + heightHalf,
+  ];
+}
```

Comment:
If you go with the boolean prop, I would move this function inside the component to avoid passing the arguments. It would also allow you to directly set `el.style.transform` to reduce duplication between the `useEffect` and the `useFrame`. (You'll just have to set `el.style.cssTest` before calling this function.)